instrumentation/log4j/log4j-context-data/log4j-context-data-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/log4j/contextdata/v2_7/SpanDecoratingContextDataInjector.java

```diff
@@ -43,6 +54,14 @@ public StringMap injectContextData(List<Property> list, StringMap stringMap) {
     newContextData.putValue(TRACE_ID, currentContext.getTraceId());
     newContextData.putValue(SPAN_ID, currentContext.getSpanId());
     newContextData.putValue(TRACE_FLAGS, currentContext.getTraceFlags().asHex());
+
+    if (BAGGAGE_ENABLED) {
```

Comment:
Can you add tests for the new feature? You can take a look at the logback implementation and do it in a similar way https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/47159e6a7ce206ab1a738fe67ab06671dc93377a/instrumentation/logback/logback-mdc-1.0/javaagent/build.gradle.kts#L15-L21

Reply:
Sorry. I am not familiar with the unit testing modules of this project. It would be great if some volunteers could provide testing cases in future. By the way, my private built agent with this feature works well for me.

Reply:
Hmm, I'll create a follow-up task for that instead.

Reply:
#8863

Reply:
Sure ~